docs/tasks.md

```diff
@@ -153,3 +139,172 @@ work has completed, spinning up replacement workers.
+
+For more information, visit the [Jobs page](https://docs.ansible.com/ansible-tower/latest/html/userguide/jobs.html) of the Ansible Tower User Guide.
+
+Below are specific details regarding each type of unified job that can be run in AWX/Tower.
+
+
+#### Run Ad Hoc Command
+
+This task spawns an `ansible` process, which then runs a command using Ansible. The different functions contained within this task do the following:
+
+- Return SSH private key data needed for the ad hoc command (only if stored in the database as `ssh_key_data`).
+- Build a dictionary of passwords for the SSH private key, SSH user and sudo/su.
+- Build an environment dictionary for Ansible.
+- Build a command line argument list for running Ansible, optionally using `ssh-agent` for public/private key authentication.
+- Return whether the task should use `bwrap`.
+
+For more information on ad hoc commands, read the [Running Ad Hoc Commands section](https://docs.ansible.com/ansible-tower/latest/html/userguide/inventories.html#running-ad-hoc-commands) of the Inventories page of the Ansible Tower User Guide.
+
+
+#### Run Job
+
+This task is a definition and set of parameters for running `ansible-playbook` via a Job Template. It defines metadata about a given playbook run, such as a named identifier, an associated inventory to run against, the project and `.yml` playbook file to run, etc.
+
+
+#### Run Project Update
+
+When a Project Update is run in AWX, an `ansible-playbook` command is composed and spawned in a pseudoterminal on one of the servers/containers that make up the AWX installation. This process runs until completion (or until a configurable timeout), and the return code, `stdout`, and `stderr` of the process are recorded in the AWX database.
+
+This task also includes a helper method to build SCM url and extra vars with parameters needed for authentication, as well as a method for returning search/replace strings to prevent output URLs from showing sensitive passwords.
+
+To read more about this topic, visit the [Projects page](https://docs.ansible.com/ansible-tower/latest/html/userguide/projects.html) of the Ansible Tower User Guide.
+
+
+#### Run Inventory Update
+
+Inventory data can be entered into AWX manually, but many users perform Inventory Syncs to import inventory data from a variety of external sources. The goal of this task is to translate the JSON inventory data returned from `ansible-inventory` into `Host`, `Group`, and `Inventory` records in the AWX database.
+
+One of the methods in this task class builds the environment dictionary for inventory import. This _used_ to be the mechanism by which any data that needed to be passed to the inventory update script was set up (since it made the inventory update job aware of its proper credentials).  However, most environment injection is currently accomplished by the credential injectors. The primary purpose of this is to point to the inventory update `INI` or `config` file.
```

Comment:
Regarding the "old" way, the `INI` files are not exclusive either either specification via environment variables, or using the credential injectors.

Talking about running inventory updates of supported types (gce, ec2, etc.) via inventory scripts, the respective credential for each of these types will lay down authentication information, usually in environment variables. Then, inventory-specific logic will lay down an INI file that is referenced from an environment variable.

The "new" way changes that we no longer use the INI file and instead use a `.yml` file which is the inventory config for the inventory plugin, and is referenced in the CLI arguments. Credentials have been rocky, but generally we have tried to keep the means of injecting the credentials mostly the same.